tiled/utils.py

```diff
@@ -487,8 +486,16 @@ def safe_json_dump_array(array):
 
 
     def default(content):
-        if isinstance(content[0], (numpy.str_, numpy.bytes_)):
-            return content.tolist()
+        # If numpy has not been imported, we can be sure that there are no
+        # instances of numpy objects, and we don't want to pay for the numpy
+        # import / dependency if it's not otherwise used.
+        if "numpy" in sys.modules:
+            import numpy
+
+            if isinstance(content[0], (numpy.str_, numpy.bytes_)):
+                return content.tolist()
+
+        return list(bytes(content))
```

Comment:
I think the proper behavior in this line is

```py
raise TypeError
```

as illustrated in [the section on `default` in the orjson docs](https://github.com/ijl/orjson#default). We want to say, "We give up; we cannot convert this object to JSON," and let the user figure it out. Converting to `bytes(...)` may fail, depending on what type `content` is, and even if it works it's not clear that the result will be sensible in this context.

Reply:
Ah okay, I did read that in the docs and tried it, but wasn't sure if we wanted it to just accept defeat :sweat_smile: Will change it to that now